nested/controllerconfig.json

```diff
@@ -0,0 +1,65 @@
+                        "[variables('scriptUri')]"
+                    ]
+                },
+                "protectedSettings":{
+                    "commandToExecute": "[variables('cmdExec')]"
+                },
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0"
+            },
+            "tags": {
+                "displayName": "install_moodle"
+            },
+            "type": "Microsoft.Compute/virtualMachines/extensions"
+        }
+    ],
+    "variables": {
+        "cmdExec": "[concat('sh ', parameters('moodleCommon').moodleInstallScriptFilename, ' ', parameters('moodleCommon').moodleVersion, ' ', concat(parameters('moodleCommon').gfsNameRoot, '0'), ' ', 'data', ' ', parameters('moodleCommon').siteURL, ' ', concat(parameters('moodleCommon').dbServerType, '-', parameters('moodleCommon').resourcesPrefix, '.', parameters('moodleCommon').dbServerType, '.database.azure.com'), ' ', parameters('moodleCommon').moodleDbName, ' ', parameters('moodleCommon').moodleDbUser, ' ', parameters('moodleCommon').moodleDbPass, ' ', parameters('moodleCommon').moodleAdminPass, ' ', concat(parameters('moodleCommon').dbLogin, '@', parameters('moodleCommon').dbServerType, '-', parameters('moodleCommon').resourcesPrefix), ' ', parameters('moodleCommon').dbLoginPassword, ' ', parameters('moodleCommon').blobStorageAccountName, ' ', parameters('blobStorageAccountKey'), ' ', parameters('moodleCommon').moodleDbUserAzure, ' ', parameters('moodleCommon').redisDns, ' ', parameters('redisPrimaryKey'), ' ', parameters('moodleCommon').elasticVm1IP, ' ', parameters('moodleCommon').installO365pluginsSwitch, ' ', parameters('moodleCommon').installElasticSearchSwitch)]",
+        "documentation01": "This sub-template applies a specific post-deployment script to the controller vm",
+        "documentation02": "It expects certain values in the 'common' datastructure.",
+        "documentation03": "    ScriptLocation              - web URI",
+        "documentation04": "    moodleInstallScriptFilename - name of script file",
+        "documentation05": "    siteURL                     - URL of the website",
+        "documentation06": "    gfsNameRoot                 - nameroot of gluster farm - note that the code applies a 0 to get to the first node",
+        "documentation07": "    jboxVmName                  - name of the controller/jumpb ox VM",
+        "documentation08": "    dbServerType                - postgres or mysql",
+        "documentation09": "    moodleDbName                - database name for moodle",
+        "documentation10": "    moodleDbUser                - database user for moodle",
+        "documentation11": "    moodleDbPass                - database password for moodleDbUser",
+        "documentation12": "    moodleAdminPass             - password for moodle admin user",
+        "scriptUri": "[concat(parameters('moodleCommon').ScriptLocation,parameters('moodleCommon').moodleInstallScriptFilename)]"
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

Reply:
I assume you meant the _artifactsLocationSasToken practice, which was done on a separate commit after this PR was issued. Marked as done as such.